File: linear-binary-search.py
from typing import List
import math

def linear_search(array:List[int], target:int)->int:
    count = 0 #contagem do numero de etapas
    for i in range(len(array)):
        count += 1
        if array[i] == target:
            return i, count
    return -1, count

def binary_search(array:List[int], target:int)->int:
    low = 0
    high = len(array)

    count = 0 #contagem do numero de etapas
    while low < high:
        count += 1
        mid = math.floor((low + high)/2)
        if array[mid] < target:
            low = mid + 1
        elif array[mid] > target:
            high = mid
        else:
            return mid, count
    return -1, count

array = [-7, -5, -1, 3, 5, 7, 11, 17, 28, 76, 85]

print(linear_search(array=array, target=11))
print(binary_search(array=array, target=11))

File: linear-binary-search.js
function linearSearch (array, target){
    let index = -1
    let count = 0
    for(let i = 0; i < array.length; i++){
        count++
        if (array[i] == target){
            // console.log(count)
            index = i;
        }
    }
    return index
}

function binarySearch (array, target){
    //ordenar array primieiro
    let firstIndex = 0
    let lastIndex = array.length - 1
    let midIndex = 0

    let count = 0

    while (lastIndex >= firstIndex){
        count ++
        midIndex = Math.floor((firstIndex + lastIndex)/2)
        if (target > array[midIndex]){
            firstIndex = midIndex + 1
        } else if (target < array[midIndex]){
            lastIndex = midIndex - 1
        } else{
            console.log(count)
            return midIndex
        }
    }
}

const array = [1, 2, 3, 4, 5, 6, 7, 8]
const target = 7
const linearIndex = linearSearch(array, target)
const binaryIndex = binarySearch(array, target)

// console.log(linearIndex)
console.log(binaryIndex)